gems/aws-sdk-core/spec/aws/shared_config_spec.rb

```diff
@@ -146,5 +146,25 @@ module Aws
 
     end
 
+    context 's3_us_east_1_regional_endpoint selection' do
+
+      it 'can resolve s3_us_east_1_regional_endpoint from config file' do
+        config = SharedConfig.new(
+          config_path: mock_config_file,
+          config_enabled: true,
+          profile_name: "s3_iad_regional"
+        )
+        expect(config.s3_us_east_1_regional_endpoint).to eq('regional')
```

Comment:
What happens with 'regional' or 'legacy' is not provided?

Reply:
Confirmed with Jason offline, will error out, will update